### PR TITLE
[Bugfix]Start Failed with param --limit-mm-per-prompt image=N for Molmo

### PR DESCRIPTION
**Pull request by congwupiece**
Fix bug  run api_server with  param  --limit-mm-per-prompt  image=N for model_type molmo
image

in function dummy_data_for_profiling .there is a assertion  num_items >= num_expected
which is always failed because in molmo model, milti-modal image is in a dict of only one item 'images'
you can find it in file molmo.py .  function  dummy_data_for_molmo 
image
this commit make num_items correct and make vllm start correct.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
According to the [supported models page](https://docs.vllm.ai/en/latest/models/supported_models.html), we don't support multi-image input for this model yet. Are you willing to update the implementation of this model?

**Comment by congwupiece** [comment]
> According to the [supported models page](https://docs.vllm.ai/en/latest/models/supported_models.html), we don't support multi-image input for this model yet. Are you willing to update the implementation of this model?

i found that this model Molmo may have already support multi-image input.  
if the vllm start success 
code below work success
```python 
    res = client.chat.completions.create(
    model=MODEL_NAME, 
    messages=[
        { 
            "role": "user", 
            "content": [{
                "type": "image_url",
                "image_url": {
                    "url": image_to_base64("case/case2.jpg")
                }
            },
            {
                "type": "image_url",
                "image_url": {
                    "url": image_to_base64("case/case1.jpg")
                }
            },
            {
                "type": "text",
                "text": "describe the pictures i show you"
            }]
        }
    ]
)
# it success return and give back 2 description
```


i think this model  has already support mliti-image .but i am not sure. related code is below .
https://github.com/vllm-project/vllm/blob/bbc3619dc806b25fd5e14eef90819052ab76e1c6/vllm/model_executor/models/molmo.py

File: vllm/inputs/registry.py [hunk 1]
```diff
@@ -244,6 +244,10 @@ def dummy_data_for_profiling(
         if dummy_data.multi_modal_data is not None:
             for k, v in dummy_data.multi_modal_data.items():
                 num_items = len(v) if isinstance(v, list) else 1
+                #fix for molmo 
+                if k == "image" and model_config.hf_config.model_type == "molmo":
```
Comment (DarkLight1337) [inline vllm/inputs/registry.py:248]: In any case, let's not hard code this here. Can you instead modify the dummy data inside the model's file?

Comment (congwupiece) [reply vllm/inputs/registry.py:248]: ok

**Pull request closed without merging**